modules/extension/complex/src/main/java/org/geotools/data/complex/feature/type/FeatureTypeRegistry.java

```diff
@@ -613,9 +613,9 @@ private AttributeType createComplexAttributeType(
             new HashMap<>();
 
     private void createFoundationTypes() {
-        Map<Name, AttributeType> foundationTypes =
-                FOUNDATION_TYPES.computeIfAbsent(helper.getClass(), o -> new HashMap<>());
-        synchronized (foundationTypes) {
+        synchronized (FOUNDATION_TYPES) {
```

Comment:
Just thinking out loud here, but why not make the class a ConcurrentHashMap instead? Is there a chance the key or value of the entries might be null?